src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -73,7 +73,7 @@ async Task AuthenticateSASL(List<string> mechanisms, bool async)
                                           "Mechanisms received from server: " + string.Join(", ", mechanisms));
             var mechanism = "SCRAM-SHA-256";
 
-            var passwd = GetPassword() ??
+            var passwd = GetPassword(GetUsername()) ??
```

Comment:
As written above, AuthenticateSASL should simply have a username parameter, just like cleartext or MD5.